core/lib/Cardano/Address/Script.hs

```diff
@@ -212,3 +296,60 @@ blake2b224 =
 -- Size, in bytes, of a hash of public key (without the corresponding chain code)
 hashSize :: Int
 hashSize = hashDigestSize Blake2b_224
+
+-- | Encode a 'KeyHash' to bech32 'Text', using @script_vkh@ as a human readable prefix.
+--
+-- @since 3.0.0
+bech32 :: KeyHash -> Text
+bech32 (KeyHash keyHash) = T.decodeUtf8 $ encode (EBech32 hrp) keyHash
+  where
+    hrp = [Bech32.humanReadablePart|script_vkh|]
+
+-- Examples of Script jsons:
+--"e09d36c79dec9bd1b3d9e152247701cd0bb860b5ebfd1de8abb6735a"
+--{ "all" : [ "e09d36c79dec9bd1b3d9e152247701cd0bb860b5ebfd1de8abb6735a"
+--          , "e09d36c79dec9bd1b3d9e152247701cd0bb860b5ebfd1de8abb6735b"
+--          ]
+--}
+--{ "all" : [ "e09d36c79dec9bd1b3d9e152247701cd0bb860b5ebfd1de8abb6735a"
+--          , {"any": [ "e09d36c79dec9bd1b3d9e152247701cd0bb860b5ebfd1de8abb6735b"
+--                    , "e09d36c79dec9bd1b3d9e152247701cd0bb860b5ebfd1de8abb6735c"
+--                    ]
+--            }
+--          ]
+--}
+--{ "all" : [ "e09d36c79dec9bd1b3d9e152247701cd0bb860b5ebfd1de8abb6735a"
+--          , {"at_least": { "from" :[ "e09d36c79dec9bd1b3d9e152247701cd0bb860b5ebfd1de8abb6735b"
+--                                   , "e09d36c79dec9bd1b3d9e152247701cd0bb860b5ebfd1de8abb6735c"
+--                                   , "e09d36c79dec9bd1b3d9e152247701cd0bb860b5ebfd1de8abb6735d"
+--                                   ]
+--                         , "m" : 2
+--                         }
+--            }
+--          ]
+--}
+
+instance ToJSON Script where
+    toJSON (RequireSignatureOf keyHash) = String $ bech32 keyHash
+    toJSON (RequireAllOf content) =
+        object ["all" .= fmap toJSON content]
+    toJSON (RequireAnyOf content) =
+        object ["any" .= fmap toJSON content]
+    toJSON (RequireMOf m content) =
+        let inside = Object ("from" .= fmap toJSON content <> "m" .= toJSON m)
+        in object ["at_least" .= inside]
+
+instance FromJSON Script where
+    parseJSON v = parseKey v <|> parseAnyOf v  <|> parseAllOf v <|> parseAtLeast v
+      where
+          parseKey = withText "Script KeyHash" $
+              either (fail . showErr) (pure . RequireSignatureOf) . keyHashFromText
+          parseAnyOf =
+              withObject "Script AnyOf" $ \o -> RequireAnyOf <$> (o .: "any" :: Json.Parser [Script])
```

Comment:
Are the type annotations needed to make it build?

Reply:
indeed - not needed - removed here https://github.com/input-output-hk/cardano-addresses/pull/84/commits/6d001e1c93efe19182d250027aaf32352ba40673